activerecord/lib/active_record/errors.rb

```diff
@@ -241,4 +241,10 @@ class ImmutableRelation < ActiveRecordError
   # The mysql, mysql2 and postgresql adapters support setting the transaction isolation level.
   class TransactionIsolationError < ActiveRecordError
   end
+
+  # UnreversableOrderError will be raised when #reverse_order can not reverse the order.
+  # ActiveRecord reverse order algorithm is not perfect and doesn't cover all possible use cases.
+  # That is why this exception can be raise when relation order is too complicated to be automatically reversed.
```

Comment:
Some of your sentences retread the same territory. How about replacing these three lines with:

```ruby
# UnreversableOrderError is raised when a relation's order is too complex for
# +reverse_order+ to automatically reverse.
```

What do you think? :smile: